packages/hardware-trezor/src/transformers/certificates.ts

```diff
@@ -0,0 +1,161 @@
+import * as Trezor from 'trezor-connect';
+import { BIP32Path } from '@cardano-sdk/crypto';
+import { Cardano } from '@cardano-sdk/core';
+import { InvalidArgumentError /* , Transform*/ } from '@cardano-sdk/util';
+import { TrezorTxTransformerContext } from '../types';
+import { stakeKeyPathFromGroupedAddress } from './keyPaths';
+
+type StakeKeyCertificateType =
+  | Trezor.CardanoCertificateType.STAKE_REGISTRATION
+  | Trezor.CardanoCertificateType.STAKE_DEREGISTRATION;
+
+type TrezorStakeKeyCertificate = {
+  type: StakeKeyCertificateType;
+  path?: BIP32Path;
+  scriptHash?: string;
+  keyHash?: string;
+};
+
+type TrezorDelegationCertificate = {
+  type: Trezor.CardanoCertificateType.STAKE_DELEGATION;
+  path?: BIP32Path;
+  scriptHash?: string;
+  pool: string;
+};
+
+type TrezorPoolRegistrationCertificate = {
+  poolParameters: Trezor.CardanoPoolParameters;
+  type: Trezor.CardanoCertificateType.STAKE_POOL_REGISTRATION;
+};
+
+const getStakeAddressCertificate = (
+  certificate: Cardano.StakeAddressCertificate,
+  context: TrezorTxTransformerContext,
+  type: StakeKeyCertificateType
+): TrezorStakeKeyCertificate => {
+  const knownAddress = context?.knownAddresses.find(
+    (address) => Cardano.RewardAccount.toHash(address.rewardAccount) === certificate.stakeKeyHash
+  );
+  const rewardAddress = knownAddress ? Cardano.Address.fromBech32(knownAddress.rewardAccount)?.asReward() : null;
+
+  let credentials;
+  if (rewardAddress?.getPaymentCredential().type === Cardano.CredentialType.KeyHash) {
+    const path = stakeKeyPathFromGroupedAddress(knownAddress);
+    credentials = path ? { path } : { keyHash: certificate.stakeKeyHash };
+  } else {
+    credentials = { scriptHash: certificate.stakeKeyHash };
+  }
+
+  return {
+    ...credentials,
+    type
+  };
+};
+
+const getStakeDelegationCertificate = (
+  certificate: Cardano.StakeDelegationCertificate,
+  context: TrezorTxTransformerContext
+): TrezorDelegationCertificate => {
+  const poolIdKeyHash = Cardano.PoolId.toKeyHash(certificate.poolId);
+  const knownAddress = context?.knownAddresses.find(
+    (address) => Cardano.RewardAccount.toHash(address.rewardAccount) === certificate.stakeKeyHash
+  );
+  const rewardAddress = knownAddress ? Cardano.Address.fromBech32(knownAddress.rewardAccount)?.asReward() : null;
+
+  let credentials;
+  if (rewardAddress?.getPaymentCredential().type === Cardano.CredentialType.KeyHash) {
+    const path = stakeKeyPathFromGroupedAddress(knownAddress);
+    credentials = path ? { path } : { keyHash: certificate.stakeKeyHash };
+  } else {
+    credentials = { scriptHash: certificate.stakeKeyHash };
+  }
```

Comment:
this conditional block to generate the credentials is duplicated between `getStakeAddressCertificate` and `getStakeDelegationCertificate`, maybe we could move that into a separate helper?